### PR TITLE
Adding -Nodejs-Logging-Novice-to-Expert article

### PR DESCRIPTION
**Pull request by ajibade3210**
Please read through the [writing guide](https://reflectoring.io/contribute/writing-guide/) before submitting a pull request and go through this checklist (at least for your first article).

**Feel free to raise the pull request even if the checklist hasn't been worked through, yet**, to take advantage of the online preview that is generated for each pull request (see "deploy/netlify" under "Show all checks" in your pull request). But please only flag it to be reviewed by your editor after going through the checklist.

## Review
- [x] you have reviewed the text a day after finishing it (you'd be surprised how many errors you can avoid by just getting some distance to what you've written)

## Content
- [x] the article explains *why* and not only *how*
- [x] (half-)sentences that express a meaningful idea that is understandable by itself own are **bold** to allow quick scanning of the text
- [x] the text has been checked with [Grammarly](https://grammarly.com) (the free tier is enough)

## Text
- [x] the text is conversational ("we" instead of "you", easy language)
- [x] there are no "walls of text" (short sentences, short paragraphs with a single idea)
- [x] the text is inclusive ("the developers" and "they" instead of "the developer" and "he/she")
- [x] the text uses active voice instead of passive voice
- [x] names are spelled consistently throughout the article (correct uppercase / lowercase)
- [x] file names and variables are highlighted as `code`

## Structure
- [x] the article has a short intro before the first headline
- [x] by reading the table of contents (TOC) alone, I get a sense of the "story" of the article (check the TOC in the preview after raising the PR or by starting the blog locally on your machine)
- [x] if using sub-headlines, there's at least two sub-headlines below a main-headline
- [x] headlines are consistent throughout the article (same level of abstraction, same style - if one headline starts with a verb in imperative form, the other headlines on the same level should probably, too)

## Code Examples
- [x] code examples are introduced with a ":" and explained below the code example
- [x] code examples are formatted to avoid horizontal scrolling (manual line breaks and 2-spaces indentation)

### CODE REVIEW

File: content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+- 3: http
+- 4: verbose
+- 5: debug
+- 6: silly
+
+Using the appropriate Log level to log, allows you to consume and route logging output based on message importance.
+
+_where:_
+
+- **Error**: is a serious problem or failure, that halts current activity but leaves the application in a recoverable state with no effect on other operations. The application can continue working
```
Comment (thombergs) [inline content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:199]: Can you combine this list with the list above? Having practically the same list two times in a row is a bit cumbersone. It should be one list, something like:

- **0 - error:** use this to log a serious problem or failure ...

File: content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+- **Warn**: A non-blocking warning about an unusual system exception. These logs provide context for a possible error. it logs warning signs that should be investigated.
+- **Info**: This denotes major events and informative messages about the application's current state. Useful For tracking the flow of the application.
+- **HTTP**: This logs out HTTP request-related messages. HTTP transactions ranging from the host, path, response, requests, etc.
+- **Verbose**: Records detailed messages that may contain sensitive information. Verbose was [renamed](https://github.com/aspnet/Announcements/issues/124) to Trace.
```
Comment (thombergs) [inline content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:203]: How is an ASP.NET log level relevant to a Node.js log level?

File: content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md (hunk 1 above)
Comment (thombergs) [inline content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:199]: Isn't there a "Critical" above "Error"?

Comment (ajibade3210) [reply content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:199]: Okay i can do this

File: content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md (hunk 2 above)
Comment (ajibade3210) [reply content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:203]: Okay i will update this

File: content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md (hunk 1 above)
Comment (ajibade3210) [reply content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:199]: here I am trying to stick with the level mentioned on the Winston README.md.
https://www.npmjs.com/package/winston#user-content-logging-levels

Should I include more level types in the list? please advise

Comment (ajibade3210) [reply content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md:199]: The default npm level used in Winston is just 0-6 which I listed already.

**Review by thombergs — CHANGES_REQUESTED** [review]
Good article! I'd love to see more articles from you :).

Sorry, I wasn't finished with the review when I posted some comments earlier ... I got interrupted. So here's the final review.

I pushed some minor editing to make the formatting and language a bit more consistent. I also left some comments on the PR for you to action. Let me know if you have any questions. Once you've done the changes, let me know and I'll hit publish!

Some general feedback:
The article was already quite good, but if you look through and edit the text a day after having written it and run it through Grammarly once, it makes a a good article to a great one :)

There were some inconsistencies in upper and lower case and formatting, but I edited that (see my changes [here](https://github.com/reflectoring/reflectoring.github.io/pull/482/commits/089aae81f3136dc63df4d203def77d94bfabfb20)). Have a look at the changes so you know what I'm looking for in the next article :).

File: content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+- maxFiles: maximum number of logs to keep. If it is not set no log will be removed. The above log is set to delete in 14 days.
+- insert `fileRotateTransport` into logger transport option.
+
+_Run `node app.js`_
+
+This generates a new `rotate-%DATE%.log` file in our `logs` folder and a `JSON()` file containing our rotate settings
+
+There are more [option settings](https://github.com/winstonjs/winston-daily-rotate-file#options) to use in the `winston-daily-rotate-file` transport.
+
+### **Winston Transports**
```
[the remark this answers is not in the document]
Comment (ajibade3210) [reply content/blog/2022/2022-07-19-Nodejs-Logging-Novice-to-Expert.md]: Thanks a lot @thombergs I learnt a lot from the things you edited.
I made some improvements to the logger level section.
Please Help me check it out

**Pull request merged**